assets/js/util/i18n.js

```diff
@@ -38,6 +39,37 @@ export const numberFormat = ( number, options = {} ) => {
 	return new Intl.NumberFormat( locale, formatOptions ).format( number );
 };
 
+/**
+ * Flattens an array using the JS Internationalization List Format API with a fallback if it is unavailable.
+ *
+ * @since n.e.x.t
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat/ListFormat|`options` parameter} For all available options.
+ *
+ * @param {Array}  list             The list to flatten.
+ * @param {Object} [options]        Formatting options.
+ * @param {string} [options.locale] Locale to use for formatting. Defaults to current locale used by Site Kit.
+ * @param {string} [options.style]  Length of the formatted message. Defaults to long.
+ * @param {string} [options.type]   Type of list. Defaults to 'conjunction' (A, B, and C).
+ *                                  Also available 'disjunction' (A, B, or C)
+ *                                  Also available 'unit' (5 pounds, 12 ounces)
+ * @return {string} The flattened list.
+ */
+export const listFlatten = ( list, options = {} ) => {
```

Comment:
I think it is a bit of a misnomer to name as `listFlatten` as this could easily be confused with functionality performed by [`_.flatten`](https://lodash.com/docs/4.17.15#flatten) which is not the case – this is more of an i18n-friendly `Array.join`.

Let's call it `listFormat` for consistency with our `numberFormat` utility which wraps `Intl.NumberFormat`.

Similarly, this utility shouldn't impose any opinionated defaults (it mostly provides the locale and passes through the rest to a new formatter and calls `format`.

I think we should have a separate `listJoin` utility if we want to have something which has more opinionated defaults.

As for the fallback, I'd rather we actually do away with that and polyfill `Intl.ListFormat` since this is likely a problem that we'll run into in other cases for other `Intl` APIs so it's worth looking into a bit more and address the root cause.